src/tactic/squeeze.lean

```diff
@@ -78,7 +147,117 @@ meta def parse_config : option pexpr → tactic (simp_config_ext × format)
      prod.mk <$> eval_expr simp_config_ext e
              <*> rec.to_tactic_format cfg
 
-meta def auto_simp_lemma := [``eq_self_iff_true]
+/-- `same_result proof tac` runs tactic `tac` and checks if the proof
+produced by `tac` is equivalent to `proof`. -/
```

Comment:
The problem here is that we're producing new proof goals and I have to check that changing the arguments of `simp` doesn't change the goals that will be produced. I think neither `is_def_eq` nor `unify` helps me with that.

Reply:
> The problem here is that we're producing new proof goals and I have to check that changing the arguments of simp doesn't change the goals that will be produced.

Now I'm even more confused.  Is there a case where `simp [...]` creates new goals?

In any case, if we want to check that the same goals are generated, then we should be comparing the goals, not the proofs.  It is a *feature* that the proof can change (and use fewer lemmas).

Reply:
Actually, now that I think about it, simp always create 0 or 1 goals which simplifies things. I was assuming that more than one goal were being generated for some reason.

Reply:
I have changed my approach. Instead of checking success by comparing proof terms, I now have a representation of goals that I can now compare in a meaningful way. And it is applicable if we want to do a `squeeze` tactic for other tactics than `simp` (e.g. `rw` which can produce multiple goals).